oauth2client/_transport/__init__.py

```diff
@@ -16,29 +16,49 @@
 
 from six.moves import http_client
 
-from oauth2client.transport import httplib2
-
-
+try:
+    from oauth2client._transport import httplib2
+except ImportError:  # pragma: NO COVER
+    httplib2 = None
+try:
+    from oauth2client._transport import urllib3
+except ImportError:  # pragma: NO COVER
+    urllib3 = None
+
+# The available transports, in order of priority.
+_TRANSPORTS = tuple(filter(None, (urllib3, httplib2,)))
+
+if not _TRANSPORTS:  # pragma: NO COVER
+    raise ImportError(
+        'No HTTP transports are available for oauth2client. Please install '
+        'urllib3 or httplib2.')
+
+_PREFERRED_TRANSPORT = _TRANSPORTS[0]
 REFRESH_STATUS_CODES = (http_client.UNAUTHORIZED,)
 
 
-class _SETTINGS(object):
-    default_transport = httplib2
+def _get_preferred_http_object(*args, **kwargs):
+    """Returns an instance of preferred transport's http object."""
+    return _PREFERRED_TRANSPORT.get_http_object(*args, **kwargs)
 
 
-def set_default_transport(transport_module):
-    """Sets the global default transport."""
-    _SETTINGS.default_transport = transport_module
+def _transport_for_http_object(http_object):
+    """Returns the transport module for a given HTTP object.
 
+    Args:
+        http_object (Any): The HTTP object.
 
-def get_default_transport():
-    """Gets the global default transport."""
-    return _SETTINGS.default_transport
+    Returns:
+        module: A transport module.
 
+    Raises:
+        ValueError: if no transport can be found for the object.
+    """
+    for transport in _TRANSPORTS:
+        if isinstance(http_object, transport.HTTP_OBJECT_CLASSES):
```

Comment:
> I don't like mutable globals.

But do you like `O(n)` lookups consistently throughout the life of the application instead of `O(1)` lookup?